### ISSUE TITLE
Hint "Functor law" is unaware of pipeline direction

### ISSUE THREAD

**Issue by andreasabel** [ndmitchell/hlint#1625]
```haskell
getContents <&> myLexer <&> pExp >>= \case
  ...
```
This triggers linter warning:
![Screenshot 2024-12-09 at 15 51 36](https://github.com/user-attachments/assets/a0305ca1-52f5-43e6-90f1-6e9ca9c5a0cc)

Applying this refactoring would create a pipeline where the flow in the middle is reversed:
```haskell
getContents <&> pExp . myLexer >>= \case
```
The pipeline in question here is: read -> lex -> parse -> case.

**Comment by zliu41** [comment]
Yeah, I imagine most people don't want this hint. I think it should be removed.

**Comment by zliu41** [comment]
The hint has been disabled in https://github.com/ndmitchell/hlint/pull/1659.
